mkosi/initrd/__main__.py

```diff
@@ -0,0 +1,102 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import argparse
+import logging
+import platform
+import shutil
+import tempfile
+
+from pathlib import Path
+from mkosi.config import OutputFormat, __version__
+from mkosi.log import log_setup
+from mkosi.run import find_binary, run, uncaught_exception_handler
+from mkosi.types import PathString
+
+
+@uncaught_exception_handler()
+def main() -> None:
+    log_setup()
+
+    parser = argparse.ArgumentParser(
+        prog="mkosi-initrd",
+        description="Build initrds or Unified Kernel Images using mkosi",
+        allow_abbrev=False,
+        usage="mkosi-initrd [options...]",
+    )
+
+    parser.add_argument("--kernel-version",
+                        metavar="KERNEL_VERSION",
+                        help="Kernel version string",
+                        default=platform.uname().release)
+    parser.add_argument("--format",
+                        choices=["cpio", "uki"],
+                        help="Output format (CPIO archive or UKI)",
+                        default="cpio")
+    parser.add_argument("--version",
+                        action="version",
+                        version=f"mkosi {__version__}")
+
+    args_initrd, args_other = parser.parse_known_args()
+
+    cmdline: list[PathString] = [
+        "mkosi",
+        "--directory", "",
+        "--format", str(args_initrd.format),
+        "--workspace-dir=/var/tmp",
+        "--package-cache-dir=/var",
```

Comment:
So doing this unconditionally means running `mkosi-initrd` as non-root is not going to work. It also means that if users pass `-fff` as the extra arguments we'll start removing stuff in /var which could be potentially dangerous.

So we should probably only do this when we're invoked as root and we should probably not allow passing arbitrary extra arguments to the script either..This means we should allow passing `--output` and `--output-dir` to this script but nothing else (unless there's an explicit use case).